### PR TITLE
feat: contact bundle with ProtoWrapper From trait

### PR DESCRIPTION
**Pull request by michaelx11**
## Overview

Issue: https://github.com/xmtp/proto/issues/81

Rebuild of #88 - this time with `ProtoWrapper` which allows us to implement `From` traits for conversion in the `vmac_protos.rs` module (happy to rename)

~~**NOTE**: I think the wasm bindings situation is currently borked. `bindings_js` and `bindings_wasm` are one cohesive operation and don't seem to build while separated (`npm test` in bindings_js complains about lack of Cargo.toml). I'll look into this next.~~

GRPC crate feature notes, copied from #88 
```
Notably - tonic does not build for wasm, so I needed to gate tonic behind a feature "grpc" inside xmtp-proto. xmtp-networking then relies on that feature. FYI @neekolas so that doesn't catch you by surprise.
```

## Tests

- Unit tests in `xmtp` core crate pass
- Fix and run bindings_js tests by adding `buf` to the Github action

### CODE REVIEW

File: xmtp/src/vmac_traits.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use xmtp_proto::xmtp::v3::message_contents::vmac_unsigned_public_key::{
```
Comment (jazzz) [inline xmtp/src/vmac_traits.rs:1]: I find this file naming interesting as there appears to be no traits in it, only Impls. 😄

File: xmtp/src/vmac.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    };
+    VmacContactBundle {
+        identity_key: Some(identity_key),
+        prekey: Some(fallback_key),
+    }
+}
+
+// Generate an outbound session (Olm Prekey Message) given a VmacContactBundle
+pub fn generate_outbound_session(bundle: VmacContactBundle) -> Vec<u8> {
+    let account = Account::new();
```
Comment (jazzz) [inline xmtp/src/vmac.rs:36]: How is this function intended to function? 

Is an account generated for every session?

File: xmtp/src/lib.rs [hunk 3]
```diff
@@ -24,4 +29,15 @@ mod tests {
...
+    fn test_can_generate_test_contact_bundle_and_session() {
+        let bundle = generate_test_contact_bundle();
+        assert!(bundle.identity_key.is_some());
+        assert!(bundle.prekey.is_some());
+
+        // Generate an outbound session (Olm Prekey Message) given a VmacContactBundle
+        let session = generate_outbound_session(bundle);
+        assert!(!session.is_empty());
+    }
 }
```
Comment (jazzz) [inline xmtp/src/lib.rs:43]: These tests could be placed in the same file as the functioned being tested. 

E.g. https://github.com/xmtp/libxmtp/blob/main/xmtp/src/builder.rs#L44

File: xmtp/src/vmac_traits.rs (hunk 1 above)
Comment (michaelx11) [reply xmtp/src/vmac_traits.rs:1]: I can rename to proto_helpers or move code back to vmac.rs? I have no strong opinions here

File: xmtp/src/vmac.rs (hunk 2 above)
Comment (michaelx11) [reply xmtp/src/vmac.rs:36]: I should prefix with `test_` - meant almost as a demo that the proto code isn't just baloney

Comment (michaelx11) [reply xmtp/src/vmac.rs]: actually moved all of the functions directly into the new `mod test` block

File: xmtp/src/lib.rs (hunk 3 above)
Comment (michaelx11) [reply xmtp/src/lib.rs]: did it!

**Comment by michaelx11** [comment]
FYI @neekolas - this provides a paradigm (one of many, probably better ones) that allows us to convert between Vodozemac types and protos. Feel free to open a PR against xmtp/proto#xmtpv3 with a completely redesigned Account proto message!

**Review by jazzz — APPROVED** [review]

**Comment by michaelx11** [comment]
@jazzz - the wrapper situation could definitely use your thoughts. I couldn't implement `From` directly between Vodozemac types and protos, because both live outside the crate (Rust orphan rules). I thought of this wrapper approach but there's almost certainly a better way.

**Comment by michaelx11** [comment]
Will ship for now to get the bindings fix (need to install buf) and the `grpc` feature flag situation in main. Very happy to revisit or completely rework the proto wrapper logic, for now it acts as a working placeholder for further protobuf-based development.

**Pull request merged**